app/experimenter/nimbus-ui/src/components/FormBranches/FormBranch.tsx

```diff
@@ -58,18 +60,26 @@ export const FormBranch = ({
   onFeatureConfigChange: (
     featureConfig: getConfig_nimbusConfig_featureConfig | null,
   ) => void;
+  defaultValues: Record<string, any>;
+  setSubmitErrors: React.Dispatch<React.SetStateAction<Record<string, any>>>;
 }) => {
   const id = fieldNamePrefix;
-
-  const {
-    register,
-    watch,
-    formState: { isSubmitted },
-  } = useFormContext();
-
   const submitErrors = { ...branch.errors };
 
+  const { register, watch } = useFormContext();
```

Comment:
You mean have `register` live inside `useCommonNestedForm`? `register` comes from `useForm` which is called in `FormBranches`, not `FormBranch`. `useCommonNestedForm` doesn't call `useForm` so it can't live inside the hook.